tests/reftests/best-effort.test

```diff
@@ -0,0 +1,13 @@
+N0REP0
+### opam switch create best-effort --empty
+### : Check that --best-effort works as intended when requesting a specific version of a package
+### <pkg:unavailable.1>
+opam-version: "2.0"
+available: false
+### <pkg:test.1>
+opam-version: "2.0"
+### <pkg:test.2>
+opam-version: "2.0"
+depends: ["unavailable"]
+### opam install --show --best-effort test.2
+Nothing to do.
```

Comment:
So I've added a simple test but the output of `opam install` doesn't seem right here.
It should have the same output as without `--best-effort`:
```
diff --git a/tests/reftests/best-effort.test b/tests/reftests/best-effort.test
index f358d5496..e30179eb8 100644
--- a/tests/reftests/best-effort.test
+++ b/tests/reftests/best-effort.test
@@ -9,5 +9,11 @@ opam-version: "2.0"
 ### <pkg:test.2>
 opam-version: "2.0"
 depends: ["unavailable"]
-### opam install --show --best-effort test.2
-Nothing to do.
+### opam install --show test.2
+[ERROR] Package conflict!
+  * Missing dependency:
+    - unavailable
+    unmet availability conditions: 'false'
+
+No solution found, exiting
+# Return code 20 #
```

Reply:
Why? You are trying to install a package that cannot be installed. Hence `--best-effort` just skips it and does nothing, while the normal command throws an error.

Reply:
Mmh, ok fair enough, I've never really used `--best-effort` ^^"

Reply:
The simplest test that I can think of is to have `my-package.2` that is unavailable and `my-package.1` that is available. Then, the output of `opam install my-package.2` and `opam install my-package.2 --best-effort` should be the same after this PR, but different before this PR.